modules/v2/main.tf

```diff
@@ -0,0 +1,285 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+resource "google_cloud_run_v2_service" "main" {
+  provider = google-beta
+
+  project     = var.project_id
+  name        = var.service_name
+  location    = var.location
+  description = var.description
+  labels      = var.service_labels
+
+  template {
+    revision        = var.revision
+    labels          = var.template_labels
+    annotations     = var.template_annotations
+    timeout         = var.timeout
+    service_account = var.service_account
+
+    execution_environment            = var.execution_environment
+    encryption_key                   = var.encryption_key
+    max_instance_request_concurrency = var.max_instance_request_concurrency
+    session_affinity                 = var.session_affinity
+
+    dynamic "scaling" {
+      for_each = var.template_scaling[*]
+      content {
+        min_instance_count = scaling.value.min_instance_count
+        max_instance_count = scaling.value.max_instance_count
+      }
+    }
+
+    dynamic "vpc_access" {
+      for_each = var.vpc_access[*]
+      content {
+        connector = vpc_access.value.connector
+        egress    = vpc_access.value.egress
+        dynamic "network_interfaces" {
+          for_each = vpc_access.value.network_interfaces[*]
+          content {
+            network    = network_interfaces.value.network
+            subnetwork = network_interfaces.value.subnetwork
+            tags       = network_interfaces.value.tags
+          }
+        }
+      }
+    }
+
+    dynamic "containers" {
+      for_each = var.containers
+      content {
+        name        = containers.value.container_name
```

Comment:
Could we use key rather than using a separate container_name?

Reply:
I tried that earlier, but having a container name is optional input and container image will be a bigger key name. Let me know your thoughts!

Reply:
Made it list(objects) instead.